tests/test_performance_codec.py

```diff
@@ -33,3 +34,59 @@ def test_encode_decode(self):
             self.assertTrue(
                 target, "The decoded Performed Part doesn't match the original."
             )
+    
+    def test_beat_normalization(self):
+        for fn in MATCH_IMPORT_EXPORT_TESTFILES: 
```

Comment:
Nice and complete tests but what do they test exactly? Are you testing the new functions and their expected values?

Please correct me if I'm wrong but from my understanding, you are still testing the `beat_period_log`, articulation, timing, and dynamics that are created with the decoding process.

Reply:
Also, tests seem to be failing.

Reply:
yes, the tests are making sure that the alternative ways of calculating beat / tempo and their respective inverse function can also reconstruct the performance. 
I can incorporate them into the `test_encode_decode` function if that's more intuitive.